ietf/doc/views_search.py

```diff
@@ -749,6 +749,10 @@ def docs_for_ad(request, name):
             if (
                 not ballot
                 or doc.get_state_slug("draft") == "repl"
+                or (
+                    doc.get_state("draft-iesg")
+                    and doc.get_state("draft-iesg").slug == "defer"
+                )
```

Comment:
You could replace this with doc.get_state_slug("draft-iesg") and not have to test to see if there is draft-iesg state. That said, there will always be draft-iesg state - that's the compromise we made to make it possible to put drafts back into iesg-state idexists.

See https://github.com/ietf-tools/datatracker/blob/f2b716fc052a0152c32b86b428ba6ebfdcdf5cd2/ietf/doc/models.py#L319-L329